app/goprotoc/free_fields.go

```diff
@@ -8,31 +8,43 @@ import (
 	"github.com/jhump/protoreflect/desc"
 )
 
-func doPrintFreeFieldNumbers(fds []*desc.FileDescriptor, w io.Writer) {
+func doPrintFreeFieldNumbers(fds []*desc.FileDescriptor, w io.Writer) error {
 	for _, fd := range fds {
 		for _, md := range fd.GetMessageTypes() {
-			printMessageFreeFields(md, w)
+			if err := printMessageFreeFields(md, w); err != nil {
+				return err
+			}
 		}
 	}
+	return nil
 }
 
-func printMessageFreeFields(md *desc.MessageDescriptor, w io.Writer) {
+func printMessageFreeFields(md *desc.MessageDescriptor, w io.Writer) error {
 	for _, nested := range md.GetNestedMessageTypes() {
 		printMessageFreeFields(nested, w)
 	}
 
 	unused := computeFreeRanges(md)
-	fmt.Fprintf(w, "%- 35s free:", md.GetFullyQualifiedName())
+	if _, err := fmt.Fprintf(w, "%- 35s free:", md.GetFullyQualifiedName()); err != nil {
```

Comment:
Ugh. This is the main reason I don't enable the `errcheck` analyzer. This makes code so unreadable, and the cases where process output encounters I/O error is so low that I usually prefer to live with the consequences of silent failure to write output. (99% of the time, if stdout fails here, stderr will fail later, too...)

Reply:
This makes me hate the error as return value pattern in Go (and lack of generics, which makes libraries with generic, concise error-handling impossible).

Reply:
It is a low chance, but if I'm testing for 100% correctness, it's worth doing/a one-time annoyance. I know it's stupid but you gotta do it if you really want to be sure.